### PR TITLE
Only accept strings and numbers as keys (#53)

### PR DESCRIPTION
**Pull request by tedpennings**
Here's the long-overdue PR from #53 ! 

We only discussed strings in #53, but I included numbers too; some tests used numbers as keys, and they seem to work just fine.

What are your thoughts on using Symbols as keys? I'm open to making changes to this PR to add in Symbol. It seems like it's probably worth the work, but I wanted to check first.

### CODE REVIEW

File: lib/lru-cache.js [hunk 1]
```diff
@@ -241,6 +253,7 @@ LRUCache.prototype.load = function (arr) {
   //A previous serialized cache has the most recent items first
   for (var l = arr.length - 1; l >= 0; l-- ) {
     var hit = arr[l]
+    typeCheckKey(hit.k)
```
Comment (isaacs) [inline lib/lru-cache.js]: This seems excessive, no?  At that point, it's already been type checked.

Comment (tedpennings) [inline lib/lru-cache.js]: Not necessarily -- the user could have dumped the data from an older version of lru-cache, or modified it manually.

Comment (isaacs) [inline lib/lru-cache.js]: Ah, good point.

**Comment by isaacs** [comment]
I changed the error string, but otherwise merged as-is.  Published as 2.7.1.  Thanks!

**Comment by tedpennings** [comment]
thanks for the merge! :dancer:

**Comment by justanotherjones** [comment]
I'd just like to point out this was a breaking API change and shouldn't really go in a patch release. While it's certainly not good practice that I was caching a few things with the key `undefined`, this caused a bit of headache for us today :(

**Comment by balinterdi** [comment]
I second @justanotherjones . This now broke our app, although we don't use `undefined` as a key *directly*. Some deeper level of dependencies do because their `lru-cache` dependencies are specified as `^2.6.5` (for example) and the app now cannot be deployed.

**Comment by isaacs** [comment]
That was always a bug and not intended to work.

If you do `cache.get(undefined)` it'll now always return `undefined` as intended, and will prevent you from setting anything using an undefined key.

I hate to break it to you that your app was already broken, but if you don't define your keys, it's better to crash than to have a cache that returns invalid data.

**Comment by balinterdi** [comment]
The problem is our app doesn't use `lru-cache` directly but through its dependencies and I doubt the problem lies with the app. A quick scan tells me `bower`, `cross-spawn-async` and `minimatch` all depend on `lru-cache`. In my case, the problem is even harder to track down as it is only produced when deploying to remote servers, not locally.

Alternatively, what could save me is being able to fix the version of `lru-cache` to `2.7.0` for the whole project, but I don't think this is possible, is it?

**Pull request closed without merging**